packages/fxa-payments-server/server/lib/flow-performance.js

```diff
@@ -0,0 +1,125 @@
+const os = require('os');
+const log = require('./logging/log')();
+
+const MAX_DATA_LENGTH = 100;
+const VERSION = 1;
+const HOSTNAME = os.hostname();
+const PERFORMANCE_TIMINGS = [
+  {
+    event: 'network',
+    timings: [
+      { from: 'redirectStart', until: 'redirectEnd' },
+      { from: 'domainLookupStart', until: 'domainLookupEnd' },
+      { from: 'connectStart', until: 'connectEnd' },
+      { from: 'responseStart', until: 'responseEnd' },
+    ],
+  },
+  {
+    event: 'server',
+    timings: [{ from: 'requestStart', until: 'responseStart' }],
+  },
+  {
+    event: 'client',
+    timings: [{ from: 'domLoading', until: 'domComplete' }],
+  },
+];
+
+function isValidTime(time, requestReceivedTime) {
+  if (typeof time !== 'number') {
+    return false;
+  }
+
+  const age = requestReceivedTime - time;
+  if (age < 0 || isNaN(age)) {
+    return false;
+  }
+
+  return true;
+}
+
+function limitLength(val) {
+  if (val && val.length > MAX_DATA_LENGTH) {
+    return val.substr(0, MAX_DATA_LENGTH);
+  }
+
+  return val;
+}
+
+module.exports = (event, request, data) => {
+  const { view, perfStartTime, flowBeginTime, flowId, navigationTiming } = data;
+
+  if (
+    !view ||
+    !perfStartTime ||
+    !flowBeginTime ||
+    !flowId ||
+    !navigationTiming
+  ) {
+    return;
+  }
+
+  // Since the flow originates in the content-server, the flow begin time should
+  // be before the perf start time.
+  if (perfStartTime < flowBeginTime) {
+    return;
+  }
+
+  const eventTime =
+    perfStartTime + event.offset + (data.requestReceivedTime - data.flushTime);
+  writePerfEvent(
+    {
+      time: eventTime,
+      flowTime: eventTime - perfStartTime,
+      type: `flow.performance.payments.${view}`,
+    },
+    data,
+    request
+  );
+
+  PERFORMANCE_TIMINGS.forEach(item => {
+    const relativeTime = item.timings.reduce((sum, timing) => {
+      const from = navigationTiming[timing.from];
+      const until = navigationTiming[timing.until];
+      if (from >= 0 && until > from) {
+        sum += until - from;
+      }
+      return sum;
+    }, 0);
+    const absoluteTime = perfStartTime + relativeTime;
+    if (
+      relativeTime > 0 &&
+      isValidTime(absoluteTime, data.requestReceivedTime)
+    ) {
+      writePerfEvent(
+        {
+          flowTime: relativeTime,
+          time: absoluteTime,
+          type: `flow.performance.payments.${view}.${item.event}`,
+        },
+        data,
+        request
+      );
+    }
+  });
+};
+
+function writePerfEvent(event, data, request) {
+  const eventData = {
```

Comment:
This event format is very subtly different from the format logged by `logFlowEvent` in content-server `lib/flow-event.js`. It appears that `country` and `region` are present there, but omitted here. Might be better to use one event format for both spots, unless there's a really good reason to omit those properties here, and extract the duplicated code out to fxa-shared.

Reply:
I think I left those out because they are not imported into Redshift.  I will add them for parity.